images/golang-aws/build.yaml

```diff
@@ -0,0 +1,6 @@
+name: golang-aws # Name of the image to be built
+
```

Comment:
Suggestion: add a variants section and tag the images with the version of Go this is based on like it's done for golang-dind image https://github.com/jetstack/testing/blob/master/images/golang-dind/build.yaml#L8 and parameterize the Go version in the Dockerfile too.